src/codewhisperer/commands/onInlineAcceptance.ts

```diff
@@ -0,0 +1,113 @@
+import { CodeWhispererCodeCoverageTracker } from '../tracker/codewhispererCodeCoverageTracker'
+import { TextEdit, WorkspaceEdit, workspace } from 'vscode'
+import { getTabSizeSetting } from '../../shared/utilities/editorUtilities'
+import { getLogger } from '../../shared/logger/logger'
+import { RecommendationHandler } from '../service/recommendationHandler'
+import { InlineCompletionService } from '../service/inlineCompletionService'
+import { sleep } from '../../shared/utilities/timeoutUtils'
+import { handleExtraBrackets } from '../util/closingBracketUtil'
+
+/**
+ * This function is called when user accepts a intelliSense suggestion or an inline suggestion
+ */
+export async function onInlineAcceptance(
+    acceptanceEntry: OnRecommendationAcceptanceEntry,
+    globalStorage: vscode.Memento
+) {
+    RecommendationHandler.instance.cancelPaginatedRequest()
+    InlineCompletionService.instance.disposeInlineCompletion()
+
+    if (acceptanceEntry.editor) {
+        sleep(CodeWhispererConstants.vsCodeCursorUpdateDelay)
+        const languageContext = runtimeLanguageContext.getLanguageContext(acceptanceEntry.editor.document.languageId)
+        const start = acceptanceEntry.range.start
+        const end = acceptanceEntry.editor.selection.active
+        const languageId = acceptanceEntry.editor.document.languageId
+        RecommendationHandler.instance.reportUserDecisionOfCurrentRecommendation(
+            acceptanceEntry.editor,
+            acceptanceEntry.acceptIndex
+        )
+        // codewhisperer will be doing editing while formatting.
+        // formatting should not trigger consoals auto trigger
+        vsCodeState.isCodeWhispererEditing = true
+        /**
+         * Mitigation to right context handling mainly for auto closing bracket use case
+         */
+        try {
+            await restoreRightContext(acceptanceEntry.editor)
+            await handleExtraBrackets(acceptanceEntry.editor, acceptanceEntry.recommendation, end)
+        } catch (error) {
+            getLogger().error(`${error} in handling right contexts`)
+        }
+        if (languageId === CodeWhispererConstants.python) {
+            await vscode.commands.executeCommand('editor.action.format')
+        } else {
+            const range = new vscode.Range(start, end)
+            const edits: TextEdit[] | undefined = await vscode.commands.executeCommand(
+                'vscode.executeFormatRangeProvider',
+                acceptanceEntry.editor.document.uri,
+                range,
+                {
+                    tabSize: getTabSizeSetting(),
+                    insertSpaces: true,
+                }
+            )
+            if (edits && acceptanceEntry.editor) {
+                const wEdit = new WorkspaceEdit()
+                wEdit.set(acceptanceEntry.editor.document.uri, edits)
+                await workspace.applyEdit(wEdit)
+            }
+        }
+        vsCodeState.isCodeWhispererEditing = false
```

Comment:
Wouldn't hurt to handle failures as `isCodeWhispererEditing` is being used as a lock:
```ts
try {
    // ...
} finally {
    vsCodeState.isCodeWhispererEditing = false
}
```